easypost/beta/referral.py

```diff
@@ -0,0 +1,132 @@
+from typing import (
+    Dict,
+    List,
+    Optional,
+)
+
+import requests
+
+from easypost.easypost_object import convert_to_easypost_object
+from easypost.requestor import (
+    RequestMethod,
+    Requestor,
+)
+
+
+class Referral:
+    @staticmethod
+    def create(
+        api_key: Optional[str] = None,
+        **params,
+    ):
+        """Create a referral user."""
+        requestor = Requestor(local_api_key=api_key)
+        new_params = {"user": params}
+        response, api_key = requestor.request(
+            method=RequestMethod.POST, url="/referral_customers", params=new_params, beta=True
+        )
+        return convert_to_easypost_object(response=response, api_key=api_key)
+
+    @staticmethod
+    def update_email(
+        email,
+        user_id,
+        api_key: Optional[str] = None,
+    ):
+        """Update a referral user."""
+        requestor = Requestor(local_api_key=api_key)
+        url = f"/referral_customers/{user_id}"
+        new_params = {
+            "user": {
+                "email": email,
+            }
+        }
+        response, api_key = requestor.request(method=RequestMethod.PUT, url=url, params=new_params, beta=True)
+        return convert_to_easypost_object(response=response, api_key=api_key)
+
+    @staticmethod
+    def all(
+        api_key: Optional[str] = None,
+        **params,
+    ) -> List:
+        """Retrieve a list of referral users from the API."""
+        requestor = Requestor(local_api_key=api_key)
+        response, api_key = requestor.request(
+            method=RequestMethod.GET, url="/referral_customers", params=params, beta=True
+        )
+        return convert_to_easypost_object(response=response, api_key=api_key)
+
+    @staticmethod
+    def add_credit_card(
+        referral_user_api_key,
+        number,
+        exp_month,
+        exp_year,
+        cvc,
+        primary_or_secondary="primary",
+    ) -> dict:
+        """Add credit card to a referral user."""
+        easypost_key = Referral._retrieve_easypost_key()
+        stripe_token = Referral._create_stripe_token(number, exp_month, exp_year, cvc, easypost_key)
+        response = Referral._create_easypost_credit_card(
+            referral_user_api_key,
+            stripe_token.get("id", ""),
```

Comment:
You may not need an empty string fallback since the default fallback is `None`.

```suggestion
            stripe_token.get("id"),
```

Reply:
We have to add a default value to pass `mypy` lint, according to this GitHub [issue](https://github.com/python/mypy/issues/9213)